src/main/java/org/opensearch/knn/index/mapper/LuceneFieldMapper.java

```diff
@@ -45,34 +44,31 @@ public class LuceneFieldMapper extends KNNVectorFieldMapper {
     static LuceneFieldMapper createFieldMapper(
         String fullname,
         Map<String, String> metaValue,
-        KNNMethodContext knnMethodContext,
-        KNNMethodConfigContext knnMethodConfigContext,
+        KNNIndexContext knnIndexContext,
+        UserProvidedParameters originalParameters,
         CreateLuceneFieldMapperInput createLuceneFieldMapperInput
     ) {
         final KNNVectorFieldType mappedFieldType = new KNNVectorFieldType(
             fullname,
             metaValue,
-            knnMethodConfigContext.getVectorDataType(),
-            new KNNMappingConfig() {
-                @Override
-                public Optional<KNNMethodContext> getKnnMethodContext() {
-                    return Optional.of(knnMethodContext);
-                }
-
-                @Override
-                public int getDimension() {
-                    return knnMethodConfigContext.getDimension();
-                }
-            }
+            () -> KNNVectorFieldType.KNNVectorFieldTypeConfig.builder()
```

Comment:
So out of curiosity, is the mapped field type persisted as part of index information? I imaging it would be else the user would have to have a mapping request everytime the the system crashes and comes up

Reply:
Its persisted via the source of the mapping. What happens is the source gets parsed into a field mapper and field type when an index is created.